pyccel/codegen/printing/ccode.py

```diff
@@ -192,6 +197,16 @@
                   ('int',1)     : 'char',
                   ('bool',4)    : 'bool'}
 
+arr_dtype_registry = {('real',8)    : 'double',
+                  ('real',4)    : 'float',
+                  ('complex',8) : 'cdouble',
+                  ('complex',4) : 'cfloat',
+                  ('int',4)     : 'int',
+                  ('int',8)     : 'long',
+                  ('int',2)     : 'sint',
+                  ('int',1)     : 'char',
+                  ('bool',4)    : 'bool'}
+
```

Comment:
Why are we using different data types for the arrays?

Reply:
I think I found why. Is this the dtype that you use in the array, that therefore cannot contain spaces? If so then why do none of these start with `nd_`?

Reply:
i should add the nd_ i did it in the printing

Reply:
Given that you never use these values without the `nd_` I don't see any reason to split it. And I think having it hear makes the difference clearer